src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -0,0 +1,3862 @@
+            summon->SetReactState(REACT_PASSIVE);
+            summon->CastSpell(summon, SPELL_BEACON_OF_HOPE_AREATRIGGER, false);
+            break;
+        }
+            break;
+
+        default:
+            break;
+        }
+    }
+
+    void SummonedCreatureDies(Creature* summon, Unit* /*killer*/) override
+    {
+        switch (summon->GetEntry())
+        {
+            case NPC_ANDUIN_DOUBT:
+            {
+
+                break;
+            }
+            case NPC_ANDUIN_DESPAIR:
+            {
+                summon->ToTempSummon()->SetTempSummonType(TEMPSUMMON_CORPSE_DESPAWN);
+                break;
+            }
+
+            case NPC_GRIM_REFLECTION:
+            {
+                summon->GetGUID();
+                summon->AI()->DoAction(ACTION_EMPOWER_REFLECTIONS);
+                summon->ToTempSummon()->SetTempSummonType(TEMPSUMMON_MANUAL_DESPAWN);
+                summon->DespawnOrUnsummon(1s);
+                break;
+            }
+
+            case NPC_LOST_SOUL:
+            {
+                summon->GetAI()->DoAction(ACTION_RESTORE_SOUL);
+            }
+
+
+            break;
+        default:
+            break;
+        }
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_ENCOUNTER_STARTED, 1);
+        instance->SetData(DATA_ANDUIN_WRYNN, IN_PROGRESS);
+        BossAI::JustEngagedWith(who);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me);
+        PhaseEvents(PHASE_ONE);
+        Talk(SAY_AGGRO);
+
+        scheduler.Schedule(10s, [this](TaskContext /*task*/)
+            {
+                if (GameObject* bridge = me->FindNearestGameObject(GAMEOBJECT_BRIDGE_TO_ANDUIN, 500.0f))
+                {
+                    bridge->SetLootState(GO_READY);
+                    bridge->UseDoorOrButton(0, false);
+                }
+            });     
+    }
+
+    void DoAction(int32 action) override
+    {
+        switch (action)
+        {
+
+        case ACTION_SUMMON_SOULS:
+        {
+            if (Creature* anduinSoul = instance->GetCreature(DATA_ANDUIN_SOUL))
+            { 
+                anduinSoul->GetAI()->DoAction(ACTION_SUMMON_KINGSMOURNE_SOULS);
+            }
+            break;
+        }
+        case ACTION_CANCEL_EVENTS:
+        {
+            events.CancelEvent(EVENT_HOPEBREAKER);
+            events.CancelEvent(EVENT_DOMINATION_WORD_PAIN);
+            events.CancelEvent(EVENT_BEFOULED_BARRIER);
+            events.CancelEvent(EVENT_KINGSMOURNE_HUNGERS);
+            events.CancelEvent(EVENT_WICKED_STAR);
+            events.CancelEvent(EVENT_BLASPHEMY);
+            events.CancelEvent(EVENT_GRIM_REFLECTIONS);
+            _dominationWordCount = 0;
+            break;
+        }
+
+        case ACTION_ARTHAS_INTERMISSION_UTHER:
+        {
+            instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_INTERMISSION, 1);
+            instance->DoUpdateWorldState(WORLD_STATE_ANDUIN_ENCOUNTER_STARTED, 0);
```

Comment:
The encounter is going on, I'm pretty sure this doesn't change back to 0 unless it's a wipe or kill.